header/header.go

```diff
@@ -132,6 +132,12 @@ func (eh *ExtendedHeader) Validate() error {
 		)
 	}
 
+	// ensure data root from raw header matches computed root
```

Comment:
why the change in order

Reply:
I will reorder those things in the next PR anyway. The order of verification matters here as well and ours is slightly off